src/mlpack/methods/edge_boxes/feature_extraction_impl.hpp

```diff
@@ -0,0 +1,1039 @@
+/**
+ * @file feature_extraction_Impl.hpp
+ * @author Nilay Jain
+ *
+ * Implementation of feature extraction methods.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_ImPL_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_ImPL_HPP
+
+
+#include "feature_extraction.hpp"
+#include <mlpack/methods/pca/pca.hpp>
+namespace mlpack {
+namespace structured_tree {
+
+/**
+ * Constructor: stores all the parameters in an object
+ * of feature_parameters class.
+ */
+template<typename MatType, typename CubeType>
+StructuredForests<MatType, CubeType>::
+StructuredForests(FeatureParameters F) 
+{
+  params = F; 
+}
+
+/*
```

Comment:
If we do not need the codes again, I think we can remove it, git will record the codes we removed before